traits/observers/_list_item_observer.py

```diff
@@ -0,0 +1,221 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.observers._list_change_event import list_event_factory
+from traits.observers._i_observer import IObserver
+from traits.observers._trait_event_notifier import TraitEventNotifier
+from traits.observers._observe import add_or_remove_notifiers
+from traits.observers._observer_change_notifier import ObserverChangeNotifier
+from traits.trait_list_object import TraitList
+
+
+@IObserver.register
+class ListItemObserver:
```

Comment:
As in the last PR, can this be a subclass of named tuple?

Also had the thought that even if you don't turn it into a named tuple, using `__slots__` here might make sense for efficiency (keep that in the back of your mind, even if it isn't part of this PR).

Reply:
I did give this a go, and while it got rid of the `__init__`, `__hash__`, `__eq__` and the properties, the equality check does not quite do what is implemented here. e.g. the following returns true.
```
ListItemObserver(notify=True, optional=False) == (True, False)
```
That is not very convenient considering that `ListItemObserver`, `SetItemObserver` and `DictItemObserver` have the same attributes and they should not be equal to each other.

Reply:
Let's keep `__slots__` away from this for now; using `__slots__` can have some interesting side-effects (particularly with regard to subclassing). It's an optimization that we should save until we need it (and my bet would be that we won't ever need it).